netboot/netconf.go

```diff
@@ -74,6 +78,91 @@ func GetNetConfFromPacketv6(d *dhcpv6.DHCPv6Message) (*NetConf, error) {
 	return &netconf, nil
 }
 
+// GetNetConfFromPacketv4 extracts network configuration information from a DHCPv4
+// Reply packet and returns a populated NetConf structure
+func GetNetConfFromPacketv4(d *dhcpv4.DHCPv4) (*NetConf, error) {
+	// extract the address from the DHCPv4 address
+	ipAddr := d.YourIPAddr()
+	if ipAddr.Equal(net.IPv4zero) {
+		return nil, errors.New("ip address is null (0.0.0.0)")
+	}
+	netconf := NetConf{}
+
+	// get the subnet mask from OptionSubnetMask. If the netmask is not defined
+	// in the packet, an error is returned
+	netmaskOption := d.GetOneOption(dhcpv4.OptionSubnetMask)
+	if netmaskOption == nil {
+		return nil, errors.New("no netmask option in response packet")
+	}
+	netmask := netmaskOption.(*dhcpv4.OptSubnetMask).SubnetMask
+	if binary.LittleEndian.Uint32(netmask) == 0 {
```

Comment:
a stronger check (i.e. with less surprises, given how Go handles IP data) could be
```
ones, bits := netmask.Size()
if bits != 32 {
    return ...
}
if ones == 0 {
    return ...
}
```

You can probably omit the `bits` check though